### PR TITLE
bug/7590-rachael-detox-github_actions_run_testRail

### PR DESCRIPTION
**Pull request by rbontrager**
## Description of Change

It was determined that the test-rail integration script was never going to run do to existing github actions logic (https://github.com/actions/runner/issues/2205).  The test-rail integration script was changed so it always runs and the testRail.yml was slightly changed so it will create a new test run if necessary. 
## Testing

- [ ] Tested on iOS 
- [ ] Tested on Android 

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: .github/workflows/e2e_android.yml [hunk 1]
```diff
@@ -119,6 +110,13 @@ jobs:
           avd-name: Pixel_4_XL_API_28
           script: yarn e2e:android-test --take-screenshots failing --updateSnapshot --headless
 
+      - name: Upload e2e-junit
+        if: failure() || success()
```
Comment (Sparowhawk) [inline .github/workflows/e2e_android.yml:114]: I don't believe you need the if statement here as it should work no matter what

File: .github/workflows/e2e_android.yml [hunk 2]
```diff
@@ -148,15 +146,9 @@ jobs:
...
-      - name: Upload e2e-junit
-        uses: actions/upload-artifact@v3
-        with:
-          name: e2e-junit
-          path: VAMobile/e2e/test_reports/e2e-junit.xml
-      
   send_test_results_to_testrail:
-    if: (github.event_name == 'workflow_run' && '!cancelled()' && github.event.workflow_run.conclusion != 'skipped') || ${{ github.event.inputs.run_testRail == true}}
-    needs: [e2e-android]
+    if: ${{ always() }}
```
Comment (Sparowhawk) [inline .github/workflows/e2e_android.yml:150]: I don't think this if statement needs to be there

File: .github/workflows/e2e_ios.yml [hunk 3]
```diff
@@ -106,6 +97,13 @@ jobs:
       - name: Run e2e tests for iOS
         run: yarn e2e:ios-test --take-screenshots failing --updateSnapshot
 
+      - name: Upload e2e-junit
+        if: failure() || success()
```
Comment (Sparowhawk) [inline .github/workflows/e2e_ios.yml:101]: same with this if statement

File: .github/workflows/e2e_ios.yml [hunk 4]
```diff
@@ -134,16 +132,10 @@ jobs:
...
-      - name: Upload e2e-junit
-        uses: actions/upload-artifact@v3
-        with:
-          name: e2e-junit
-          path: VAMobile/e2e/test_reports/e2e-junit.xml
 
   send_test_results_to_testrail:
-    if: (github.event_name == 'workflow_run' && '!cancelled()' && github.event.workflow_run.conclusion != 'skipped') || ${{ github.event.inputs.run_testRail == true}}
-    needs: [e2e-ios]
+    if: ${{ always() }}
```
Comment (Sparowhawk) [inline .github/workflows/e2e_ios.yml:137]: same with this if statement

**Review by Sparowhawk — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e_ios.yml (hunk 4 above)
Comment (rbontrager) [reply .github/workflows/e2e_ios.yml:137]: This is necessary. If you take a look at the link I have in the description, there is currently some github actions logic that will cause it not to run this job without that statement when start_slack_job is skipped (ie when we run it manually).

**Review by Sparowhawk — DISMISSED** [review]

File: .github/workflows/e2e_android.yml [hunk 5]
```diff
@@ -148,15 +155,9 @@ jobs:
...
-        uses: actions/upload-artifact@v3
-        with:
-          name: e2e-junit
-          path: VAMobile/e2e/test_reports/e2e-junit.xml
-      
   send_test_results_to_testrail:
-    if: (github.event_name == 'workflow_run' && '!cancelled()' && github.event.workflow_run.conclusion != 'skipped') || ${{ github.event.inputs.run_testRail == true}}
-    needs: [e2e-android]
+    if: always() && (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion != 'skipped') || github.event.inputs.run_testRail == 'true'
+    needs: e2e-android
```
Comment (theodur) [inline .github/workflows/e2e_android.yml:160]:
```suggestion
    needs: e2e-android
```

File: .github/workflows/e2e_ios.yml [hunk 6]
```diff
@@ -9,17 +9,8 @@ on:
     branches:
       - 'dependabot/**'
   workflow_dispatch:
-    inputs:
-      run_testRail:
-        description: Run_TestRail
-        type: boolean
-        required: false
```
Comment (theodur) [inline .github/workflows/e2e_ios.yml:16]: Looks like this was accidentally removed?

File: .github/workflows/e2e_ios.yml [hunk 7]
```diff
@@ -9,17 +9,8 @@ on:
...
   schedule:
     - cron: '0 4 * * 1,2,3,4,5'
-  workflow_run:
-    workflows: ['[Release] New Release Issue']
-    types: 
-      - completed
```
Comment (theodur) [inline .github/workflows/e2e_ios.yml:22]: Same with this trigger

Comment (rbontrager) [reply .github/workflows/e2e_ios.yml]: Added back in

File: .github/workflows/e2e_ios.yml (hunk 6 above)
Comment (rbontrager) [reply .github/workflows/e2e_ios.yml]: Added back in

**Review by theodur — APPROVED** [review]

**Pull request merged**